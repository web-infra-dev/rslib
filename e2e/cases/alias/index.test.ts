import { expect, test } from 'vitest';
import { buildAndGetResults } from '#shared';

test('alias should work', async () => {
  const fixturePath = __dirname;
  const { entries } = await buildAndGetResults(fixturePath);

  expect(entries.esm).toContain('hello world');
  expect(entries.cjs).toContain('hello world');

  // simple artifacts check
  expect(entries.esm).toMatchSnapshot();
  expect(entries.cjs).toMatchSnapshot();
});
