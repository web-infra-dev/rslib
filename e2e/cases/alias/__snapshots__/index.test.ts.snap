// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`alias should work 1`] = `
"var __webpack_modules__ = ({});
/************************************************************************/
// The module cache
var __webpack_module_cache__ = {};

// The require function
function __webpack_require__(moduleId) {

// Check if module is in cache
var cachedModule = __webpack_module_cache__[moduleId];
if (cachedModule !== undefined) {
return cachedModule.exports;
}
// Create a new module (and put it into the cache)
var module = (__webpack_module_cache__[moduleId] = {
exports: {}
});
// Execute the module function
__webpack_modules__[moduleId](module, module.exports, __webpack_require__);

// Return the exports of the module
return module.exports;

}

/************************************************************************/
// webpack/runtime/rspack_version
(() => {
__webpack_require__.rv = function () {
	return "1.0.0-canary-0b368b6-20240704171208";
};

})();
// webpack/runtime/rspack_unique_id
(() => {
__webpack_require__.ruid = "bundler=rspack@1.0.0-canary-0b368b6-20240704171208";

})();
/************************************************************************/
var __webpack_exports__ = {};

;// CONCATENATED MODULE: ./e2e/cases/alias/src/a.ts
const a = 'hello world';

;// CONCATENATED MODULE: ./e2e/cases/alias/src/index.ts

console.info(a);

export { a };
"
`;

exports[`alias should work 2`] = `
"(() => { // webpackBootstrap
"use strict";
var __webpack_modules__ = ({});
/************************************************************************/
// The module cache
var __webpack_module_cache__ = {};

// The require function
function __webpack_require__(moduleId) {

// Check if module is in cache
var cachedModule = __webpack_module_cache__[moduleId];
if (cachedModule !== undefined) {
return cachedModule.exports;
}
// Create a new module (and put it into the cache)
var module = (__webpack_module_cache__[moduleId] = {
exports: {}
});
// Execute the module function
__webpack_modules__[moduleId](module, module.exports, __webpack_require__);

// Return the exports of the module
return module.exports;

}

/************************************************************************/
// webpack/runtime/rspack_version
(() => {
__webpack_require__.rv = function () {
	return "1.0.0-canary-0b368b6-20240704171208";
};

})();
// webpack/runtime/rspack_unique_id
(() => {
__webpack_require__.ruid = "bundler=rspack@1.0.0-canary-0b368b6-20240704171208";

})();
/************************************************************************/
var __webpack_exports__ = {};

;// CONCATENATED MODULE: ./e2e/cases/alias/src/a.ts
const a = 'hello world';

;// CONCATENATED MODULE: ./e2e/cases/alias/src/index.ts

console.info(a);

var __webpack_export_target__ = exports;
for(var i in __webpack_exports__) __webpack_export_target__[i] = __webpack_exports__[i];
if(__webpack_exports__.__esModule) Object.defineProperty(__webpack_export_target__, '__esModule', { value: true });
})()
;"
`;
